src/mapblock_mesh.cpp

```diff
@@ -1215,7 +1216,7 @@ MapBlockMesh::MapBlockMesh(MeshMakeData *data, v3s16 camera_offset):
 		// Create meshbuffer
 		// This is a "Standard MeshBuffer",
 		// it's a typedeffed CMeshBuffer<video::S3DVertex>
-		scene::SMeshBuffer *buf = new scene::SMeshBuffer();
+		scene::SMeshBufferTangents *buf = new scene::SMeshBufferTangents();
```

Comment:
Tangents buffers have to be default when shaders used. its not case of specific setting and thats not gonna change no matter what. Parallax mapping is just example why it cannot be done correctly before. Anything else wasnt possible just because of missing tangent space. Shaders were trying to emulate that (absurdly costly calculations on GPU side per each vertex) but it was only partially correct. Whole mapblock mesh when using shaders HAVE to use tangents no matter the effect settings. Also remember that shaders are instanced, apart from having to fall back to faulty and slowly emulation when using regular vertex, it would mean double number of shaders instances.
And slowdown was 1fps without shaders. With shaders enabled using tangent space calculated by enginge not the shaders results with like 30% more fps.